C/CURL/build_tarballs.jl

```diff
@@ -0,0 +1,82 @@
+# Copied from `L/LibCURL/build_tarballs.jl`
+using BinaryBuilder, Pkg
+
+name = "CURL"
+version = v"7.78.0"
+
+# Collection of sources required to build LibCURL
+sources = [
+    ArchiveSource("https://curl.haxx.se/download/curl-$(version).tar.gz",
+                  "ed936c0b02c06d42cf84b39dd12bb14b62d77c7c4e875ade022280df5dcc81d7"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/curl-*
+
+# Holy crow we really configure the bitlets out of this thing
+FLAGS=(
+    # Disable....almost everything
+    --without-ssl --without-gnutls --without-gssapi
+    --without-libidn --without-libidn2 --without-librtmp
+    --without-nss --without-polarssl
+    --without-spnego --without-libpsl --disable-ares --disable-manual
+    --disable-ldap --disable-ldaps --without-zsh-functions-dir
+    --disable-static --disable-libgsasl
+
+    # A few things we actually enable
+    --with-libssh2=${prefix} --with-zlib=${prefix} --with-nghttp2=${prefix}
+    --enable-versioned-symbols
+)
+
+
+if [[ ${target} == *mingw* ]]; then
+    # We need to tell it where to find libssh2 on windows
+    FLAGS+=(LDFLAGS="${LDFLAGS} -L${prefix}/bin")
+
+    # We also need to tell it to link against schannel (native TLS library)
+    FLAGS+=(--with-schannel)
+elif [[ ${target} == *darwin* ]]; then
+    # On Darwin, we need to use SecureTransport (native TLS library)
+    FLAGS+=(--with-secure-transport)
+
+    # We need to explicitly request a higher `-mmacosx-version-min` here, so that it doesn't
+    # complain about: `Symbol not found: ___isOSVersionAtLeast`
+    if [[ "${target}" == aarch64* ]]; then
+        export CFLAGS=-mmacosx-version-min=11.0
+    else
+        export CFLAGS=-mmacosx-version-min=10.11
+    fi
```

Comment:
The aarch64 branch for sure isn't needed anymore.  Not sure about the x86_64 one, might still be needed (by default it'd be `-mmacosx-version-min=10.10`)